qutebrowser/extensions/interceptors.py

```diff
@@ -29,6 +30,34 @@
     from PyQt5.QtCore import QUrl
 
 
+class ResourceType(enum.Enum):
+    """Possible request types that can be received.
+
+    Currently corresponds to the QWebEngineUrlRequestInfo Enum:
+    https://doc.qt.io/qt-5/qwebengineurlrequestinfo.html#ResourceType-enum
+    """
+
+    MAIN_FRAME = 1
+    SUB_FRAME = 2
+    STYLESHEET = 3
+    SCRIPT = 4
+    IMAGE = 5
+    FONT_RESOURCE = 6
+    SUB_RESOURCE = 7
+    OBJECT = 8
+    MEDIA = 9
+    WORKER = 10
+    SHARED_WORKER = 11
+    PREFETCH = 12
+    FAVICON = 13
+    XHR = 14
+    PING = 15
+    SERVICE_WORKER = 16
+    CSP_REPORT = 17
+    PLUGIN_RESOURCE = 18
+    UNKNOWN = 19
```

Comment:
I'm putting this up for others to see:

> < beelzebielsk> jgkamat: I'm wondering why there's a need for an
> enumeration of the resource types, and a mapping between Qt resource
> types and QB resource types. I'm also wondering why there are QB
> resource types at all, since they're just the list of Qt resource
> types.
> 
> < toofar> beelzebielsk: Partly because qutebrowser supports multiple
> backends (and even if webkit support goes away it'll likely stay
> achitected that way) and also because we have no control of the
> backend APIs so it is nice to have an abstraction layer so if they
> change we only have to change on one place 
> 
> < jgkamat> Yup, the idea is that if qt makes qtwebskynet or whatever the api can stay the same
> 
> < jgkamat> Although, abp has a similar api, maybe we should restrict the types to those

Reply:
I agree those should be mapped (I'd rather not expose QtWebEngine stuff directly via the extension API). I also prefer exposing them all, not just the abp ones.

However, those should be lower-case for consistency with enums elsewhere. Yeah, we use all-upper for constants usually, but all other enum members (see e.g. `usertypes.py`) are lower-case.